zipline/pipeline/engine.py

```diff
@@ -348,7 +348,10 @@ def run_chunked_pipeline(self,
             # if we don't have to.
             return chunks[0]
 
-        return categorical_df_concat(chunks, inplace=True)
+        # Filter out empty chunks. Empty dataframes lose dtype information,
+        # which makes concatenation fail.
+        nonempty_chunks = [c for c in chunks if len(c)]
+        return categorical_df_concat(nonempty_chunks, inplace=True)
```

Comment:
One argument I could see here is that `categorical_df_concat` should just always do this filtering. I'm not confident that that's always the right behavior for other consumers though, so I left this here. If we end up adding similar filtering for other consumers, we should probably move this into concat.